README.md

```diff
@@ -1,2 +1,423 @@
 # web-eid-authtoken-validation-dotnet
-Web eID authentication token validation library for .NET
+
+![European Regional Development Fund](https://raw.githubusercontent.com/open-eid/DigiDoc4-Client/master/client/images/EL_Regionaalarengu_Fond.png)
+
+*web-eid-authtoken-validation-dotnet* is a .NET library for issuing challenge nonces and validating Web eID JWT authentication tokens during secure authentication with electronic ID (eID) smart cards in web applications.
+
+More information about the Web eID project is available on the project [website](https://web-eid.eu/).
+
+# Quickstart
+
+Complete the steps below to add support for secure authentication with eID cards to your ASP.NET Core web application back end. Instructions for the front end are available [here](https://github.com/web-eid/web-eid.js).
+
+See full example [here](https://github.com/web-eid/web-eid-asp-dotnet-example).
+
+## 1. Add the library to your project
+To install the package, you can use the Package Manager Console. 
+1.  Select the  **Tools**  >  **NuGet Package Manager**  >  **Package Manager Console**  menu command.
+2.  Once the console opens, check that the  **Default project**  drop-down list shows the project into which you want to install the package. If you have a single project in the solution, it is already selected.
+3. Enter the command `Install-Package Newtonsoft.Json`. The console window shows output for the command.
+
+When you install a package, NuGet records the dependency in either your project file or a `packages.config` file (depending on the project format).
+
+## 2. Add cache support
+
+The validation library needs a cache for storing issued challenge nonces. 
+For this purpose generic ICache interface is provided. 
+We use [MemoryCache](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.caching.memorycache) as implementation.
+Implement the cache as follows:
+```cs
+using System;
+using System.Runtime.Caching;
+using WebEid.Security.Cache;
+
+internal sealed class MemoryCache<T> : ICache<T>
+{
+    private readonly MemoryCache cacheContent = new MemoryCache("test-cache");
+    private readonly CacheItemPolicy cacheItemPolicy;
+
+    public MemoryCache() : this(DateTimeOffset.MaxValue) { }
+
+    private MemoryCache(DateTimeOffset cacheItemExpiration)
+    {
+        this.cacheItemPolicy = new CacheItemPolicy { AbsoluteExpiration = cacheItemExpiration };
```

Comment:
We should use [`SlidingExpirationPolicy`](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.caching.cacheitempolicy.slidingexpiration) and a default time span of 5 minutes. Does not matter much though when we migrate to version 2 of Web eID.